tensorflow_model_optimization/python/core/clustering/keras/cluster_wrapper.py

```diff
@@ -221,30 +222,41 @@ def build(self, input_shape):
           shape=pulling_indices.shape,
           dtype=tf.int32,
           trainable=False,
+          synchronization=tf.VariableSynchronization.ON_READ,
+          aggregation=tf.VariableAggregation.ONLY_FIRST_REPLICA,
           initializer=initializers.Constant(
               value=k.batch_get_value([pulling_indices])[0]
           )
       )
 
       # We store these pairs to easily update this variables later on
-      self.clustered_vars.append((weight_name, weight))
+      self.ori_weights_vars_tf[weight_name] = self.add_weight(
+          'ori_weights_vars_tf',
+          shape=weight.shape,
+          dtype=weight.dtype,
+          trainable=True,
+          initializer=initializers.Constant(
+              value=k.batch_get_value([weight])[0]
+          )
+      )
 
     # We use currying here to get an updater which can be triggered at any time
     # in future and it would return the latest version of clustered weights
     def get_updater(for_weight_name):
       def fn():
-        return self.clustering_impl[for_weight_name].get_clustered_weight(
-            self.pulling_indices_tf[for_weight_name]
-        )
+        # Get the clustered weights
+        pulling_indices = self.pulling_indices_tf[for_weight_name]
+        clustered_weights = self.clustering_impl[for_weight_name].\
+            get_clustered_weight(pulling_indices)
+        return clustered_weights
 
       return fn
 
     # This will allow us to restore the order of weights later
     # This loop stores pairs of weight names and how to restore them
-
     for ct, weight in enumerate(self.layer.weights):
       name = self._weight_name(weight.name)
-      full_name = self.layer.name + "/" + name
+      full_name = '{}{}{}'.format(self.layer.name, '/', name)
```

Comment:
the character '/' could be inside of the string

Reply:
Thanks, done.